### PR TITLE
Fix 2016.3 test suite hangs

### PR DESCRIPTION
**Pull request by s0undt3ch**
### What does this PR do?

Prevents the tests suite from hanging on 2016.3
### Previous Behavior

The test suite would hang and not finish, at least on 2016.3
### New Behavior

Should no longer hang.
### Tests written?
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by s0undt3ch** [comment]
Go Go Jenkins!

File: tests/unit/modules/boto_iot_test.py [hunk 1]
```diff
@@ -98,6 +98,7 @@ def _has_required_boto():
                       ruleDisabled=True)
 
 
+@skipIf(True, 'These tests are pegging the CPU and eating all available RAM')
 class BotoIoTTestCaseBase(TestCase):
```
Comment (s0undt3ch) [inline tests/unit/modules/boto_iot_test.py]: @kraney would you mind revisiting these to see what's going wrong?
You can trigger this by running all unittests `python tests/runtests.py -u`.

Comment (jtand) [inline tests/unit/modules/boto_iot_test.py]: @kraney would you also be able to look at the other test failures that are here? It looks like when the initial PR was made, we didn't have boto3 being put on the test boxes, so these tests never actually ran.

**Comment by kraney** [comment]
Hm, I am able to reproduce by running all of the unit tests as you say - sort of, because it actually failed in modules/boto_apigateway_test rather than in IoT. But If i run the individual unit test, it runs fine.

`./tests/runtests.py -n unit.modules.boto_iot_test`

MagicMock seems to be running differently when running the full suite. Where this call should throw an exception, it doesn't, and things go in a tight loop making the call over and over thinking there are always more results available.

```
self.conn.get_api_keys.side_effect = ClientError(error_content, 'get_api_keys')
result = boto_apigateway.describe_api_keys(**conn_parameters)
```

**Comment by jtand** [comment]
@kraney You mean it fails by hanging? I am seeing that on branch tests as well, but they seem to all be related. Disabling these as @s0undt3ch has the apigateway_test didn't hang.

**Comment by rallytime** [comment]
These changes were pulled in with #31926. Closing this one.

**Comment by s0undt3ch** [comment]
Anytime!

**Pull request closed without merging**